pagopa/pn-frontend
# PN-3379-FE-errore-invio-notifica-senza-pagamento

## Short description
In the manual notification creation, there was a scenario involving the case in which no payment documents (modelli di pagamento) are to be attached, in which the body sent to the API call for the creation of the notification was malformed, and hence the API rejected the operation. I detected that this problem could be solved by enhancing the way we manage the information about attached files during the edition.

## List of changes proposed in this pull request
- Enhance the way payment documents attached' info is handled in the Redux store when the user go back from "Modelli di pagamento", now such information shouldn't include file structures which doesn't actually point to a file.

## How to test
Retest at least the particular scenario which fails: set "Nessuno" as "Modello di pagamento" in the first step, proceed until the last step "Modelli di pagamento"(which states that no documents should be attached), then go to the precedent step through the "Torna a Allegati" button, then return to the "Modelli di pagamento" step, and finally press "Invia". The body of the API call should not include payment info for any of the recipients, and if no other errors are present, the notification should be added.
It would be advisable to test with at least one case of Persona Fisica and one case of Persona Giuridica as recipients.
Then it would be also advisable to play thoroughly with the navigation in the stepper, going back and forward with different scenarios regarding the files attached in the "Allegati" and "Modelli di pagamento" steps.

Besides the body of the API call, the evolution of the Redux slice newNotification could be verified as well.
